sht3x/sht3x.c

```diff
@@ -94,16 +95,27 @@ int16_t sht3x_read(int32_t *temperature, int32_t *humidity) {
 
 int16_t sht3x_probe(void) {
     uint16_t status;
-    return sensirion_i2c_delayed_read_cmd(
-        SHT3X_ADDRESS, SHT3X_CMD_READ_STATUS_REG, SHT3X_CMD_DURATION_USEC,
-        &status, 1);
+    return sensirion_i2c_delayed_read_cmd(SHT3X_ADDRESS,
+                                          SHT3X_CMD_READ_STATUS_REG,
+                                          SHT3X_CMD_DURATION_USEC, &status, 1);
 }
 
 void sht3x_enable_low_power_mode(uint8_t enable_low_power_mode) {
     sht3x_cmd_measure =
         enable_low_power_mode ? SHT3X_CMD_MEASURE_LPM : SHT3X_CMD_MEASURE_HPM;
 }
 
+int16_t sht3x_read_serial(uint32_t *serial) {
+    int16_t ret;
+
+    ret = sensirion_i2c_delayed_read_cmd(
+        SHT3X_ADDRESS, SHT3X_CMD_READ_SERIAL_ID, SHT3X_CMD_DURATION_USEC,
+        (uint16_t *)serial, SENSIRION_NUM_WORDS(*serial));
+    SENSIRION_WORDS_TO_BYTES(serial, SENSIRION_NUM_WORDS(*serial));
+    *serial = be32_to_cpu(*serial);
```

Comment:
These conversions from bytes to u16, to bytes and then to u32 seems a bit strange. Should we add a read bytes version to embedded-common? We have this at several places, right? Also the constant swapping makes the code hard to understand and verify.

Maybe just read to a `u16t serial_number_words[2]` and then `*serial =(serial_number_words[0]<<16) | serial_number_words[1];`, which avoids some of the byte shuffling.

Reply:
that works for one word, but it's two words, so it'd need two more shifts. I'd rather stick to `be32_to_cpu` which should make everything above clear(er)